### ISSUE TITLE
Laravel 9 support

### ISSUE THREAD

**Issue by mlobalobal** [PlatformCommunity/flysystem-bunnycdn#17]
Hi, 

Thanks for this package.

Will this package be upgraded for laravel 9 ? 

Thanks!

**Comment by mlobalobal** [comment]
coool, thanks!

**Comment by mlobalobal** [comment]
got this error when trying to upload file to bunny

`Class PlatformCommunity\Flysystem\BunnyCDN\BunnyCDNAdapter contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (League\Flysystem\FilesystemAdapter::directoryExists)`

**Comment by mlobalobal** [comment]
```php
Storage::extend('bunnycdn', function ($app, $config) {
    $adapter = new BunnyCDNAdapter(
        new BunnyCDNStorage(
            $config['storage_zone'],
            $config['api_key'],
            $config['region']
        )
    );

    return new FilesystemAdapter(
        new Filesystem($adapter, $config),
        $adapter,
        $config
    );
})
```

change the `Storage::extend` to use FilesystemAdapter from laravel 9 upgrade guide
but when uploading, nothing happening, and no file get uploaded to bunny storage,

**Comment by sifex** [comment]
Let me give this a check :)

**Comment by sifex** [comment]
@mlobalobal after losing my mind a little on this one, I re-wrote the base client to better adhere to FlySystem's base test cases (`FilesystemAdapterTestCase`). 

tldr; My fork of https://github.com/BunnyWay/BunnyCDN.PHP.Storage was far too hard to manage long term.

You can try out `v3` which should now work perfectly with Laravel v9 – I tested it out on a new instance and it seems to work fine. It should target the `v3` / `v3.x-dev` branch automatically.


```bash
composer require platformcommunity/flysystem-bunnycdn "^3"
```

**Important:**
The one implementation difference is that you now need to swap out "BunnyCDNClient" with an instance of "BunnyCDNClient". Parameter locations should be the same. 

So the full Laravel Provider should look like:

```php
Storage::extend('bunnycdn', function ($app, $config) {
    $adapter = new BunnyCDNAdapter(
        new BunnyCDNClient(    # The bit that changed.
            $config['storage_zone'],
            $config['api_key'],
            $config['region']
        ),
        'http://testlkjhlhaslfkh19082f12f.b-cdn.net'
    );

    return new FilesystemAdapter(
        new Filesystem($adapter, $config),
        $adapter,
        $config
    );
});
```

**Comment by sifex** [comment]
Closing issue now. Please feel free to let me know if you're still coming across any issues.
